MaskRCNN/model_fpn.py

```diff
@@ -285,7 +285,7 @@ def generate_fpn_proposals_batch_tf_op(multilevel_anchor_boxes,
     all_boxes = []
     all_scores = []
     if cfg.FPN.PROPOSAL_MODE == 'Level':
-        fpn_nms_topk = cfg.RPN.TRAIN_PER_LEVEL_NMS_TOPK if training else cfg.RPN.TEST_PER_LEVEL_NMS_TOPK
+        fpn_nms_topk = cfg.RPN.TRAIN_PER_LEVEL_NMS_TOPK*batch_size if training else cfg.RPN.TEST_PER_LEVEL_NMS_TOPK
```

Comment:
Are you sure this is right? You pick top-k per image, not top-k per batch?

Reply:
The `rois` that we return here are pooled across the images in the batch, so I think it makes sense to scale the number of ROIs with batch size. When I did the tests with duplicate image, the fastrcnn box loss is properly normalized only when I do this scaling.

Reply:
i.e., this line: https://github.com/armandmcqueen/tensorpack-mask-rcnn/blob/432ad8072efedb880f4a66260b4c81919e7fd5e2/MaskRCNN/model_fpn.py#L370

Reply:
Okay - that's different than what PyTorch does, but they have a TODO note about changing the code to do top-k per image so I think it's fine.

Reply:
Okay - that's different than what PyTorch does, but they have a TODO note about changing the code to do top-k per image so I think it's fine.